x-pack/legacy/plugins/siem/public/utils/route/types.ts

```diff
@@ -55,4 +59,6 @@ export type SpyRouteProps = RouteComponentProps<{
   tabName: HostsTableType | undefined;
   search: string;
   flowTarget: FlowTarget | undefined;
-}>;
+}> & {
+  state?: Record<string, string | undefined>;
```

Comment:
is it a good idea to allow a Record with key string with an undefined value. I feel like if the value is undefined we should not add the key in the object.

Reply:
https://github.com/elastic/kibana/pull/55173/files/9ac7b15d9bd942c3ddf040ece037cd980b9ea000#diff-f0513e4cfb90441e54375f10a9f0157aR350
in case we didn't allow for undefined then in a case like this we would have to have additional logic to don't pass anything or if only one of the values would be undefined then we would need to filter those keys out, so I think it will be easier to work with to allow undefined values

Reply:
or we can just do that `<SpyRoute state={{ ruleName: rule?.name ?? 'unknown' }} />` || `<SpyRoute state={{ ruleName: rule?.name ?? '' }} />`

Reply:
yeah, but then we need to have additional logic to handle empty strings or 'unknown' string case